packages/preview-middleware-client/src/adp/controllers/AddFragment.controller.ts

```diff
@@ -59,6 +63,28 @@ export default class AddFragment extends BaseDialog {
         this.dialog.open();
     }
 
+    private specialIndexHandling(specialIndexAggregation: string | number): void {
+        const overlay = OverlayRegistry.getOverlay(this.runtimeControl as UI5Element);
+        const aggregations = overlay.getDesignTimeMetadata().getData().aggregations;
+
+        if(specialIndexAggregation in aggregations && 'specialIndexHandling' in aggregations[specialIndexAggregation]) {
+            this.model.setProperty('/indexHandlingFlag', false);
+            this.model.setProperty('/specialIndexHandlingIcon', true);
+        } else {
+            this.model.setProperty('/indexHandlingFlag', true);
+            this.model.setProperty('/specialIndexHandlingIcon', false);
+            this.model.setProperty('/specialIndexHandlingIconPressed', false);
+        }
+    }
+
+    onSpecialIndexIconPress(oEvent: Event): void  {
+        const icon = oEvent.getSource() as Icon;
+        const controlType = this.runtimeControl.getMetadata().getName();
+
+        this.popover.getContent()[0].setProperty('text', `Index is defined by special logic of ${controlType} and can't be set here`);
```

Comment:
When you put the Popover control inside `*view.xml` you will no longer need a variable here. You can get the control byId. And in the test, you will only need to mock controller.byId method without having to access private variables like this: addFragment['popover'].

Reply:
resolved in [e94a899](https://github.com/SAP/open-ux-tools/pull/1477/commits/e94a89919304cadb80bf98ae1eb4ece1069cc8cb)